cart/cart.py

```diff
@@ -1,100 +1,81 @@
 from decimal import Decimal
-
 from django.conf import settings
-
 from shipping.calculator import get_book_shipping_cost
 from store.models import Product
 
-
 class Cart:
     def __init__(self, request):
         """Initialize the cart."""
         self.session = request.session
+        cart_data = self.session.get(settings.CART_SESSION_ID)
+        self.cart_data = cart_data if cart_data is not None else {}
 
-        cart = self.session.get(settings.CART_SESSION_ID)
-
-        if not cart:
+        if not self.cart_data:
             # save an empty cart in the session
-            cart = self.session[settings.CART_SESSION_ID] = {}
-
-        self.cart = cart
+            self.cart_data = self.session[settings.CART_SESSION_ID] = {}
 
     def add(self, product, quantity=1, update_quantity=False):
         """Add a product to the cart or update its quantity."""
         product_id = str(product.id)
-
-        if product_id not in self.cart:
-            self.cart[product_id] = {
+        if product_id not in self.cart_data:
+            self.cart_data[product_id] = {
                 "product_title": product.title,
                 "product_id": product_id,
                 "quantity": 0,
                 "price": str(product.price),
             }
         if update_quantity:
-            self.cart[product_id]["quantity"] = quantity
+            self.cart_data[product_id]["quantity"] = quantity
         else:
-            self.cart[product_id]["quantity"] += quantity
-
+            self.cart_data[product_id]["quantity"] += quantity
         self.save()
 
     def save(self):
```

Comment:
Since this has a side effect, let's add a test to make sure it works as expected.